File: moderate/reverse_add.rb
File.open(ARGV[0]).each_line do |line|
  a, r, i = line.to_i, 0, 0
  while i < 100
    r = a.to_s.reverse.to_i
    break if a == r
    a, i = a+r, i+1
  end
  puts a == r ? i.to_s + " " + a.to_s : "not found"
end

File: moderate/reverse_add.py
import fileinput

for line in fileinput.input():
    a = int(line)
    b = 0

    while b < 1000 and str(a) != str(a)[::-1]:
        a += int(str(a)[::-1])
        b += 1
    print str(b) + ' ' + str(a) if b < 1000 else "not found"
